### PR TITLE
feat(sdk): support `PipelineTaskFinalStatus` in tasks that use `.ignore_upstream_failure()`

### PR DESCRIPTION
**Pull request by connor-mccarthy**
**Description of your changes:**
Support `PipelineTaskFinalStatus` in task that use `.ignore_upstream_failure()`. For example:

```python
from kfp import dsl
from kfp.dsl import PipelineTaskFinalStatus

@dsl.component
def worker_component() -> str:
    return 'hello'

@dsl.component
def cancel_handler(
    status: PipelineTaskFinalStatus,
    text: str = '',
) -> str:
    print(text)
    print(status)

@dsl.pipeline
def my_pipeline(string: str = 'string'):
    worker_task = worker_component()
    exit_task = cancel_handler(text=t.output).ignore_upstream_failure()
```

**Checklist:**
- [x] The title for your pull request (PR) should follow our title convention. [Learn more 
about the pull request title convention used in this repository](https://github.com/kubeflow/pipelines/blob/master/CONTRIBUTING.md#pull-request-title-convention).

### CODE REVIEW

File: sdk/python/kfp/compiler/pipeline_spec_builder.py [hunk 1]
```diff
@@ -1328,6 +1329,41 @@ def build_spec_by_group(
...
+    pipeline_task_spec: pipeline_spec_pb2.PipelineTaskSpec,
+):
+    if task._ignore_upstream_failure_tag:
+        pipeline_task_spec.trigger_policy.strategy = (
+            pipeline_spec_pb2.PipelineTaskSpec.TriggerPolicy.TriggerStrategy
+            .ALL_UPSTREAM_TASKS_COMPLETED)
+
+        for input_name, input_spec in (task.component_spec.inputs or
+                                       {}).items():
+            if type_utils.is_task_final_status_type(input_spec.type):
```
Comment (chensun) [inline sdk/python/kfp/compiler/pipeline_spec_builder.py:1343]: nit: consider reducing the level of nesting via
```python
if not ...
     continue
```

**Review by chensun — COMMENTED** [review]
/lgtm

File: sdk/python/kfp/compiler/pipeline_spec_builder.py (hunk 1 above)
Comment (connor-mccarthy) [reply sdk/python/kfp/compiler/pipeline_spec_builder.py]: Thanks! Done.

**Review by chensun — APPROVED** [review]
/lgtm
/approve 

Thanks

**Pull request merged**